### PR TITLE
Add log path for unsupported block log version exception

### PR DESCRIPTION
**Pull request by allenhan2**
## Change Description

For issue: https://github.com/EOSIO/eos/issues/9871

## Change Type
**Select *ONE*:**
- [ ] Documentation

- [ ] Stability bug fix

- [ ] Other

- [x] Other - special case

Test:

warn  2021-01-19T17:37:30.340 thread-0  chain_plugin.cpp:1321         plugin_initialize    ] 3190001 block_log_unsupported_version: unsupported version of block log
Unsupported version of block log. Block log version is 808464432 while code supports version(s) [1,4], log file: /Users/yue.han/workspace/eosio/develop/bin/dogfooding/data/blocks/blocks.log
    {"version":808464432,"min":1,"max":4,"log":"/Users/yue.han/workspace/eosio/develop/bin/dogfooding/data/blocks/blocks.log"}
    thread-0  block_log.cpp:57 read_from

## Testing Changes
**Select *ANY* that apply:**
- [ ] New Tests

- [ ] Existing Tests

- [ ] Test Framework

- [ ] CI System

- [x] Other

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: libraries/chain/block_log.cpp [hunk 1]
```diff
@@ -279,7 +281,10 @@ namespace eosio { namespace chain {
    public:
 
      block_log_data() = default;
-     block_log_data(const fc::path& path, mapmode mode = mapmode::readonly) { open(path, mode); }
+     block_log_data(const fc::path& path, mapmode mode = mapmode::readonly) { 
+        preamble.log_path = path;
+        open(path, mode); 
+     }
 
      const block_log_preamble& get_preamble() const { return preamble; }
```
Comment (heifner) [inline libraries/chain/block_log.cpp:290]: `open` is public but does not set the `log_path`
Should either be made private or set `log_path`.

File: libraries/chain/block_log.cpp [hunk 2]
```diff
@@ -297,6 +302,7 @@ namespace eosio { namespace chain {
       uint32_t      first_block_num() const { return preamble.first_block_num; }
       uint64_t      first_block_position() const { return first_block_pos; }
       chain_id_type chain_id() const { return preamble.chain_id(); }
+      fc::path      log_path() const { return preamble.log_path; }
```
Comment (heifner) [inline libraries/chain/block_log.cpp:305]: This is not used.

**Review by heifner — CHANGES_REQUESTED** [review]

File: libraries/chain/block_log.cpp (hunk 1 above)
Comment (allenhan2) [reply libraries/chain/block_log.cpp:290]: It is in constructor, preamble is struct

Comment (heifner) [reply libraries/chain/block_log.cpp:290]: But if you call it `open` again with a diff log_path they will be out of sync.

Comment (allenhan2) [reply libraries/chain/block_log.cpp:290]: I see the point, thanks, it is better to update in open also.

**Review by heifner — APPROVED** [review]

**Pull request merged**